kobo/apps/trackers/utils.py

```diff
@@ -45,15 +52,65 @@ def update_nlp_counter(
     kwargs = {}
     if service.endswith('asr_seconds'):
         kwargs['total_asr_seconds'] = F('total_asr_seconds') + amount
-        if asset_id is not None:
-            # If we're not updating the catch-all counter, increment any NLP add-ons the user may have
-            PlanAddOn.increment_add_ons_for_user(user_id, 'seconds', amount)
+        if asset_id is not None and organization is not None:
+            handle_usage_increment(organization, 'seconds', amount)
     if service.endswith('mt_characters'):
         kwargs['total_mt_characters'] = F('total_mt_characters') + amount
-        if asset_id is not None:
-            PlanAddOn.increment_add_ons_for_user(user_id, 'character', amount)
+        if asset_id is not None and organization is not None:
+            handle_usage_increment(organization, 'character', amount)
 
     NLPUsageCounter.objects.filter(pk=counter_id).update(
         counters=IncrementValue('counters', keyname=service, increment=amount),
         **kwargs,
     )
+
+
+@cache_for_request
+def get_organization_usage(organization: Organization, usage_type: UsageType) -> int:
+    """
+    Get the used amount for a given organization and usage type
+    """
+    usage_calc = ServiceUsageCalculator(
+        organization.owner.organization_user.user, organization
+    )
+    usage_calc._clear_cache()  # Do not use cached values
+    usage = usage_calc.get_cached_usage(USAGE_LIMIT_MAP[usage_type])
+
+    return usage
+
+
+def get_organization_remaining_usage(
+    organization: Organization, usage_type: UsageType
+) -> Union[int, None]:
+    """
+    Get the organization remaining usage count for a given limit type
+    """
+    PlanAddOn = apps.get_model('stripe', 'PlanAddOn')  # noqa
+
+    plan_limit = get_organization_plan_limit(organization, usage_type)
+    usage = get_organization_usage(organization, usage_type)
+    addon_limit, addon_remaining = PlanAddOn.get_organization_totals(
+        organization,
+        usage_type,
+    )
+    remaining = addon_limit + plan_limit - usage
```

Comment:
Shouldn't we be using addon_remaining here?

Reply:
Yes,  that's right! 

Please validate this example case for me: User A on Organization X has used 1000 minutes for NLP transcription this month, a while ago they bought 1 add on with 2000 minutes that has 1000 remaining because they used 500 on the past month and 500 this month, their plan has 500 minutes per month. 

So if their usage is 1000, that means they went beyond their plan limit and therefore used the add ons. So the remaining usage should only include the add on remaining. But if their usage was 300, for instance, that would mean they have not passed their plan limit and therefore they still have 200 plan minutes left. And that should be summed to the add on remaining. 

Is this correct?

Reply:
I believe so. Let me write it out in another way so we can make sure we are on the same page.

Organization X is on the community plan. They get 10 transcription minutes per month. In September they bought an addon for 1,000 transcription minutes. That month they used 510 minutes. Therefore, their remaining addon limit at the end of September was 500 minutes.

They use 10 minutes of transcription on October 1. This gets taken from their monthly limit, so the remaining addon limit stays at 500. On October 2, they use 200 minutes. Their addon remaining limit goes down to 300. On October 3, they use 300 minutes. Their addon remaining limit goes down to 0.

Reply:
Ok, good, the change I made for this handles this logic. It checks if there are remaining minutes in the plan and if there are it takes that into account,  if there are still more minutes to be used it takes them from the add ons.